### ISSUE TITLE
jx boot fails on step create values

### ISSUE THREAD

**Issue by rowlandwatkins** [jenkins-x/jx#6608]
### Summary 

Running jx boot to install Jenkins X serverless on an existing EKS cluster fails when Helm attempts to convert env/parameters.yaml to JSON. Any tips on moving forward highly appreciated!

### Steps to reproduce the behavior

```
jx boot
```

### Expected behavior

Jenkins X serverless is made available on EKS cluster

### Actual behavior

Install crashes and stops:

```
generated schema file /home/blah/blah/jenkins-x-boot-config/env/parameters.schema.json from template /home/blah/blah/jenkins-x-boot-config/env/parameters.tmpl.schema.json
DEBUG: Verifying Registry...

Jenkins X Admin Username admin [Automatically accepted existing value]
Jenkins X Admin Password *** [Automatically accepted existing value]
Pipeline bot Git username username [Automatically accepted existing value]
Pipeline bot Git email address user@somedomain.com [Automatically accepted existing value]
Pipeline bot Git token *** [Automatically accepted existing value]
HMAC token, used to validate incoming webhooks. Press enter to use the generated token *** [Automatically accepted existing value]
Do you want to configure an external Docker Registry? false [Automatically accepted existing value]
error: error enabling access to an external docker registry: unmarshaling /home/blah/blah/jenkins-x-boot-config/env/parameters.yaml: error converting YAML to JSON: yaml: line 1: did not find expected node content
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step create values --name parameters' command in directory 'env', output: ''
```

env/parameters.yaml:
```
adminUser:
  password: local:jenkinsx-cluster/adminUser:password
  username: admin
enableDocker: false
pipelineUser:
  email: user@somedomain.com
  token: local:jenkinsx-cluster/pipelineUser:token
  username: user
prow:
  hmacToken: local:jenkinsx-cluster/prow:hmacToken
```

### Jx version

The output of `jx version` is:

```
jx                 2.0.1150
Kubernetes cluster v1.14.9-eks-c0eccc
kubectl            v1.17.2
helm client        v2.12.2+g7d2b0c7
helm server        v2.12.2+g7d2b0c7
git                2.20.1
Operating System   Ubuntu 19.04
```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

EKS created by eksctl

### Operating system / Environment
Ubuntu 19.04

**Comment by daveconde** [comment]
@rowlandwatkins Is the above your full unmodified contents of your parameters file?

**Comment by rowlandwatkins** [comment]
Hi @daveconde, yes env/parameters.yaml is unmodified, autogenerated by jx (I'm assuming it's Heml that's getting upset by this file).

Cheers,

Rowland

**Comment by deanesmith** [comment]
@rowlandwatkins at first glance that version of Helm is quite a ways back.  2.16.1 is what is recommended.  Otherwise, not sure what would be causing this.

**Comment by rowlandwatkins** [comment]
Hi @deanesmith, ah, jx appears to have me on Helm 2.12.2 - let me see what I can do to upgrade Helm then get back to you!

**Comment by rowlandwatkins** [comment]
Ok, so I upgraded Helm and Tiller to 2.16.1:

```
Client: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

But I still get the same error - should I try trashing my local jx install and starting again?

**Comment by rowlandwatkins** [comment]
Ok, I've figured it out - this has nothing to do with Helm, it appears to be purely a go-yaml parsing problem (at least in my case). 

go-yaml doesn't seem to like any value starting with a ',' (or similar). In my case, the jenkins x admin password began with a ',' - when it attempted to unmarshal in:

pkg/cmd/step/create/step_create_values.go:192

pkg/helm/helm_helper.go performs the unmarshalling and borks when it hit my password in parameters.yaml (localsecrets get injected at runtime, so I only saw the final parameters.yaml output after building my own custom jx and adding extra debugging). After changing the password stored in local secrets, I no longer see this error.

So in summary: YAML values starting with special characters may upset go-yaml

**Comment by deanesmith** [comment]
Thank you @rowlandwatkins.  I'm not entirely sure this is easily repairable.  It would be helpful to capture this information as a caveat (NOTE) in [this document](https://jenkins-x.io/docs/getting-started/setup/boot/clouds/amazon/), though I am not sure the appropriate section.  WDYT?
